From a5af17707590d687a50deae7465f0bd1b3e45753 Mon Sep 17 00:00:00 2001
From: Lei Xu <lei@apache.org>
Date: Wed, 23 Sep 2015 10:31:09 -0700
Subject: [PATCH 0838/1023] Revert "CLOUDEEA-BUILD. Shade Netty 4."

This reverts commit b1b5e14e1ee9c6d8e8b22a0900f119fd4f93b987.

Change-Id: I4231bc8159ca51fc4272b4de1a5c163dc62af5ce

Conflicts:
	hadoop-common-project/cloudera-netty/pom.xml
---
 hadoop-client/pom.xml                              |   16 -------
 hadoop-common-project/hadoop-nfs/pom.xml           |   11 -----
 hadoop-common-project/pom.xml                      |    1 -
 hadoop-hdfs-project/hadoop-hdfs-nfs/pom.xml        |    5 --
 hadoop-hdfs-project/hadoop-hdfs/pom.xml            |   12 ++---
 .../server/datanode/web/DatanodeHttpServer.java    |   26 +++++-----
 .../datanode/web/SimpleHttpProxyHandler.java       |   42 ++++++++--------
 .../hdfs/server/datanode/web/URLDispatcher.java    |    8 ++--
 .../datanode/web/webhdfs/ExceptionHandler.java     |   20 ++++----
 .../server/datanode/web/webhdfs/HdfsWriter.java    |   16 +++----
 .../datanode/web/webhdfs/ParameterParser.java      |    2 +-
 .../datanode/web/webhdfs/WebHdfsHandler.java       |   50 ++++++++++----------
 .../tools/offlineImageViewer/FSImageHandler.java   |   30 ++++++------
 .../tools/offlineImageViewer/WebImageViewer.java   |   28 +++++------
 .../datanode/web/webhdfs/TestParameterParser.java  |    2 +-
 .../hadoop-mapreduce-client/pom.xml                |   11 -----
 hadoop-mapreduce-project/pom.xml                   |    4 --
 hadoop-project/pom.xml                             |   10 ++--
 18 files changed, 120 insertions(+), 174 deletions(-)

diff --git a/hadoop-client/pom.xml b/hadoop-client/pom.xml
index 6fb49d6..710f8d5 100644
--- a/hadoop-client/pom.xml
+++ b/hadoop-client/pom.xml
@@ -199,10 +199,6 @@
           <groupId>io.netty</groupId>
           <artifactId>netty</artifactId>
         </exclusion>
-        <exclusion>
-          <groupId>org.apache.hadoop</groupId>
-          <artifactId>cloudera-netty-all</artifactId>
-        </exclusion>
       </exclusions>
     </dependency>
 
@@ -251,10 +247,6 @@
           <groupId>io.netty</groupId>
           <artifactId>netty</artifactId>
         </exclusion>
-        <exclusion>
-          <groupId>org.apache.hadoop</groupId>
-          <artifactId>cloudera-netty-all</artifactId>
-        </exclusion>
       </exclusions>
     </dependency>
 
@@ -303,10 +295,6 @@
           <groupId>io.netty</groupId>
           <artifactId>netty</artifactId>
         </exclusion>
-        <exclusion>
-          <groupId>org.apache.hadoop</groupId>
-          <artifactId>cloudera-netty-all</artifactId>
-        </exclusion>
       </exclusions>
     </dependency>
 
@@ -335,10 +323,6 @@
           <groupId>io.netty</groupId>
           <artifactId>netty</artifactId>
         </exclusion>
-        <exclusion>
-          <groupId>org.apache.hadoop</groupId>
-          <artifactId>cloudera-netty-all</artifactId>
-        </exclusion>
       </exclusions>
     </dependency>
 
diff --git a/hadoop-common-project/hadoop-nfs/pom.xml b/hadoop-common-project/hadoop-nfs/pom.xml
index 760a1e9..af426e4 100644
--- a/hadoop-common-project/hadoop-nfs/pom.xml
+++ b/hadoop-common-project/hadoop-nfs/pom.xml
@@ -83,17 +83,6 @@
       <scope>runtime</scope>
     </dependency>
     <dependency>
-      <groupId>org.apache.hadoop</groupId>
-      <artifactId>cloudera-netty-all</artifactId>
-      <exclusions>
-        <exclusion>
-          <groupId>io.netty</groupId>
-          <artifactId>netty-all</artifactId>
-        </exclusion>
-      </exclusions>
-      <scope>compile</scope>
-    </dependency>
-    <dependency>
       <groupId>io.netty</groupId>
       <artifactId>netty</artifactId>
       <scope>compile</scope>
diff --git a/hadoop-common-project/pom.xml b/hadoop-common-project/pom.xml
index 9f9c67f..9ec4032 100644
--- a/hadoop-common-project/pom.xml
+++ b/hadoop-common-project/pom.xml
@@ -38,7 +38,6 @@
     <module>hadoop-nfs</module>
     <module>hadoop-minikdc</module>
     <module>hadoop-kms</module>
-    <module>cloudera-netty</module>
   </modules>
 
   <build>
diff --git a/hadoop-hdfs-project/hadoop-hdfs-nfs/pom.xml b/hadoop-hdfs-project/hadoop-hdfs-nfs/pom.xml
index 4b5327d..66681f8 100644
--- a/hadoop-hdfs-project/hadoop-hdfs-nfs/pom.xml
+++ b/hadoop-hdfs-project/hadoop-hdfs-nfs/pom.xml
@@ -47,11 +47,6 @@ http://maven.apache.org/xsd/maven-4.0.0.xsd">
       <scope>provided</scope>
     </dependency>
     <dependency>
-      <groupId>org.apache.hadoop</groupId>
-      <artifactId>cloudera-netty-all</artifactId>
-      <scope>compile</scope>
-    </dependency>
-    <dependency>
       <groupId>io.netty</groupId>
       <artifactId>netty</artifactId>
       <scope>compile</scope>
diff --git a/hadoop-hdfs-project/hadoop-hdfs/pom.xml b/hadoop-hdfs-project/hadoop-hdfs/pom.xml
index de2ac84..8032494 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/pom.xml
+++ b/hadoop-hdfs-project/hadoop-hdfs/pom.xml
@@ -181,19 +181,13 @@ http://maven.apache.org/xsd/maven-4.0.0.xsd">
       <scope>compile</scope>
     </dependency>
     <dependency>
-      <groupId>org.apache.hadoop</groupId>
-      <artifactId>cloudera-netty-all</artifactId>
-      <exclusions>
-        <exclusion>
-          <groupId>io.netty</groupId>
-          <artifactId>netty-all</artifactId>
-        </exclusion>
-      </exclusions>
+      <groupId>io.netty</groupId>
+      <artifactId>netty</artifactId>
       <scope>compile</scope>
     </dependency>
     <dependency>
       <groupId>io.netty</groupId>
-      <artifactId>netty</artifactId>
+      <artifactId>netty-all</artifactId>
       <scope>compile</scope>
     </dependency>
     <dependency>
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/DatanodeHttpServer.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/DatanodeHttpServer.java
index 86648f6..f645c96 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/DatanodeHttpServer.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/DatanodeHttpServer.java
@@ -17,19 +17,19 @@
  */
 package org.apache.hadoop.hdfs.server.datanode.web;
 
-import com.cloudera.io.netty.bootstrap.ChannelFactory;
-import com.cloudera.io.netty.bootstrap.ServerBootstrap;
-import com.cloudera.io.netty.channel.ChannelFuture;
-import com.cloudera.io.netty.channel.ChannelInitializer;
-import com.cloudera.io.netty.channel.ChannelPipeline;
-import com.cloudera.io.netty.channel.EventLoopGroup;
-import com.cloudera.io.netty.channel.nio.NioEventLoopGroup;
-import com.cloudera.io.netty.channel.socket.SocketChannel;
-import com.cloudera.io.netty.channel.socket.nio.NioServerSocketChannel;
-import com.cloudera.io.netty.handler.codec.http.HttpRequestDecoder;
-import com.cloudera.io.netty.handler.codec.http.HttpResponseEncoder;
-import com.cloudera.io.netty.handler.ssl.SslHandler;
-import com.cloudera.io.netty.handler.stream.ChunkedWriteHandler;
+import io.netty.bootstrap.ChannelFactory;
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.handler.codec.http.HttpRequestDecoder;
+import io.netty.handler.codec.http.HttpResponseEncoder;
+import io.netty.handler.ssl.SslHandler;
+import io.netty.handler.stream.ChunkedWriteHandler;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.apache.hadoop.conf.Configuration;
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/SimpleHttpProxyHandler.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/SimpleHttpProxyHandler.java
index 6a2e6c9..ffa7681 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/SimpleHttpProxyHandler.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/SimpleHttpProxyHandler.java
@@ -17,31 +17,31 @@
  */
 package org.apache.hadoop.hdfs.server.datanode.web;
 
-import com.cloudera.io.netty.bootstrap.Bootstrap;
-import com.cloudera.io.netty.buffer.Unpooled;
-import com.cloudera.io.netty.channel.Channel;
-import com.cloudera.io.netty.channel.ChannelFuture;
-import com.cloudera.io.netty.channel.ChannelFutureListener;
-import com.cloudera.io.netty.channel.ChannelHandlerContext;
-import com.cloudera.io.netty.channel.ChannelInboundHandlerAdapter;
-import com.cloudera.io.netty.channel.ChannelInitializer;
-import com.cloudera.io.netty.channel.ChannelPipeline;
-import com.cloudera.io.netty.channel.SimpleChannelInboundHandler;
-import com.cloudera.io.netty.channel.socket.SocketChannel;
-import com.cloudera.io.netty.channel.socket.nio.NioSocketChannel;
-import com.cloudera.io.netty.handler.codec.http.DefaultFullHttpRequest;
-import com.cloudera.io.netty.handler.codec.http.DefaultHttpResponse;
-import com.cloudera.io.netty.handler.codec.http.HttpRequest;
-import com.cloudera.io.netty.handler.codec.http.HttpRequestEncoder;
-import com.cloudera.io.netty.handler.codec.http.HttpResponseEncoder;
+import io.netty.bootstrap.Bootstrap;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.handler.codec.http.DefaultFullHttpRequest;
+import io.netty.handler.codec.http.DefaultHttpResponse;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpRequestEncoder;
+import io.netty.handler.codec.http.HttpResponseEncoder;
 import org.apache.commons.logging.Log;
 
 import java.net.InetSocketAddress;
 
-import static com.cloudera.io.netty.handler.codec.http.HttpHeaders.Names.CONNECTION;
-import static com.cloudera.io.netty.handler.codec.http.HttpHeaders.Values;
-import static com.cloudera.io.netty.handler.codec.http.HttpResponseStatus.INTERNAL_SERVER_ERROR;
-import static com.cloudera.io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
+import static io.netty.handler.codec.http.HttpHeaders.Names.CONNECTION;
+import static io.netty.handler.codec.http.HttpHeaders.Values;
+import static io.netty.handler.codec.http.HttpResponseStatus.INTERNAL_SERVER_ERROR;
+import static io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
 
 /**
  * Dead simple session-layer HTTP proxy. It gets the HTTP responses
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/URLDispatcher.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/URLDispatcher.java
index b169665..ff3f468 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/URLDispatcher.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/URLDispatcher.java
@@ -17,10 +17,10 @@
  */
 package org.apache.hadoop.hdfs.server.datanode.web;
 
-import com.cloudera.io.netty.channel.ChannelHandlerContext;
-import com.cloudera.io.netty.channel.ChannelPipeline;
-import com.cloudera.io.netty.channel.SimpleChannelInboundHandler;
-import com.cloudera.io.netty.handler.codec.http.HttpRequest;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.HttpRequest;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.hdfs.server.datanode.web.webhdfs.WebHdfsHandler;
 
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/ExceptionHandler.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/ExceptionHandler.java
index a46fdc1..fea40d7 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/ExceptionHandler.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/ExceptionHandler.java
@@ -19,9 +19,9 @@
 
 import com.sun.jersey.api.ParamException;
 import com.sun.jersey.api.container.ContainerException;
-import com.cloudera.io.netty.buffer.Unpooled;
-import com.cloudera.io.netty.handler.codec.http.DefaultFullHttpResponse;
-import com.cloudera.io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.HttpResponseStatus;
 import org.apache.commons.logging.Log;
 import org.apache.hadoop.hdfs.web.JsonUtil;
 import org.apache.hadoop.ipc.RemoteException;
@@ -32,13 +32,13 @@
 import java.io.FileNotFoundException;
 import java.io.IOException;
 
-import static com.cloudera.io.netty.handler.codec.http.HttpHeaders.Names.CONTENT_LENGTH;
-import static com.cloudera.io.netty.handler.codec.http.HttpHeaders.Names.CONTENT_TYPE;
-import static com.cloudera.io.netty.handler.codec.http.HttpResponseStatus.BAD_REQUEST;
-import static com.cloudera.io.netty.handler.codec.http.HttpResponseStatus.FORBIDDEN;
-import static com.cloudera.io.netty.handler.codec.http.HttpResponseStatus.INTERNAL_SERVER_ERROR;
-import static com.cloudera.io.netty.handler.codec.http.HttpResponseStatus.NOT_FOUND;
-import static com.cloudera.io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
+import static io.netty.handler.codec.http.HttpHeaders.Names.CONTENT_LENGTH;
+import static io.netty.handler.codec.http.HttpHeaders.Names.CONTENT_TYPE;
+import static io.netty.handler.codec.http.HttpResponseStatus.BAD_REQUEST;
+import static io.netty.handler.codec.http.HttpResponseStatus.FORBIDDEN;
+import static io.netty.handler.codec.http.HttpResponseStatus.INTERNAL_SERVER_ERROR;
+import static io.netty.handler.codec.http.HttpResponseStatus.NOT_FOUND;
+import static io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
 import static org.apache.hadoop.hdfs.server.datanode.web.webhdfs.WebHdfsHandler.APPLICATION_JSON;
 
 class ExceptionHandler {
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/HdfsWriter.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/HdfsWriter.java
index dcaeb22..0433ce6 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/HdfsWriter.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/HdfsWriter.java
@@ -17,12 +17,12 @@
  */
 package org.apache.hadoop.hdfs.server.datanode.web.webhdfs;
 
-import com.cloudera.io.netty.channel.ChannelFutureListener;
-import com.cloudera.io.netty.channel.ChannelHandlerContext;
-import com.cloudera.io.netty.channel.SimpleChannelInboundHandler;
-import com.cloudera.io.netty.handler.codec.http.DefaultHttpResponse;
-import com.cloudera.io.netty.handler.codec.http.HttpContent;
-import com.cloudera.io.netty.handler.codec.http.LastHttpContent;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.DefaultHttpResponse;
+import io.netty.handler.codec.http.HttpContent;
+import io.netty.handler.codec.http.LastHttpContent;
 import org.apache.commons.logging.Log;
 import org.apache.hadoop.hdfs.DFSClient;
 import org.apache.hadoop.io.IOUtils;
@@ -30,8 +30,8 @@
 import java.io.IOException;
 import java.io.OutputStream;
 
-import static com.cloudera.io.netty.handler.codec.http.HttpHeaders.Names.CONNECTION;
-import static com.cloudera.io.netty.handler.codec.http.HttpHeaders.Values.CLOSE;
+import static io.netty.handler.codec.http.HttpHeaders.Names.CONNECTION;
+import static io.netty.handler.codec.http.HttpHeaders.Values.CLOSE;
 
 class HdfsWriter extends SimpleChannelInboundHandler<HttpContent> {
   private final DFSClient client;
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/ParameterParser.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/ParameterParser.java
index 20dac99..0ebf3dc 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/ParameterParser.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/ParameterParser.java
@@ -17,7 +17,7 @@
  */
 package org.apache.hadoop.hdfs.server.datanode.web.webhdfs;
 
-import com.cloudera.io.netty.handler.codec.http.QueryStringDecoder;
+import io.netty.handler.codec.http.QueryStringDecoder;
 import org.apache.commons.io.Charsets;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.permission.FsPermission;
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/WebHdfsHandler.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/WebHdfsHandler.java
index 6ea3646..b60049c 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/WebHdfsHandler.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/WebHdfsHandler.java
@@ -18,17 +18,17 @@
 package org.apache.hadoop.hdfs.server.datanode.web.webhdfs;
 
 import com.google.common.base.Preconditions;
-import com.cloudera.io.netty.buffer.Unpooled;
-import com.cloudera.io.netty.channel.ChannelFutureListener;
-import com.cloudera.io.netty.channel.ChannelHandlerContext;
-import com.cloudera.io.netty.channel.SimpleChannelInboundHandler;
-import com.cloudera.io.netty.handler.codec.http.DefaultFullHttpResponse;
-import com.cloudera.io.netty.handler.codec.http.DefaultHttpResponse;
-import com.cloudera.io.netty.handler.codec.http.HttpHeaders;
-import com.cloudera.io.netty.handler.codec.http.HttpMethod;
-import com.cloudera.io.netty.handler.codec.http.HttpRequest;
-import com.cloudera.io.netty.handler.codec.http.QueryStringDecoder;
-import com.cloudera.io.netty.handler.stream.ChunkedStream;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.DefaultHttpResponse;
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.QueryStringDecoder;
+import io.netty.handler.stream.ChunkedStream;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.apache.hadoop.conf.Configuration;
@@ -56,20 +56,20 @@
 import java.security.PrivilegedExceptionAction;
 import java.util.EnumSet;
 
-import static com.cloudera.io.netty.handler.codec.http.HttpHeaders.Names.ACCESS_CONTROL_ALLOW_METHODS;
-import static com.cloudera.io.netty.handler.codec.http.HttpHeaders.Names.ACCESS_CONTROL_ALLOW_ORIGIN;
-import static com.cloudera.io.netty.handler.codec.http.HttpHeaders.Names.CONNECTION;
-import static com.cloudera.io.netty.handler.codec.http.HttpHeaders.Names.CONTENT_LENGTH;
-import static com.cloudera.io.netty.handler.codec.http.HttpHeaders.Names.CONTENT_TYPE;
-import static com.cloudera.io.netty.handler.codec.http.HttpHeaders.Names.LOCATION;
-import static com.cloudera.io.netty.handler.codec.http.HttpHeaders.Values.CLOSE;
-import static com.cloudera.io.netty.handler.codec.http.HttpMethod.GET;
-import static com.cloudera.io.netty.handler.codec.http.HttpMethod.POST;
-import static com.cloudera.io.netty.handler.codec.http.HttpMethod.PUT;
-import static com.cloudera.io.netty.handler.codec.http.HttpResponseStatus.CONTINUE;
-import static com.cloudera.io.netty.handler.codec.http.HttpResponseStatus.CREATED;
-import static com.cloudera.io.netty.handler.codec.http.HttpResponseStatus.OK;
-import static com.cloudera.io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
+import static io.netty.handler.codec.http.HttpHeaders.Names.ACCESS_CONTROL_ALLOW_METHODS;
+import static io.netty.handler.codec.http.HttpHeaders.Names.ACCESS_CONTROL_ALLOW_ORIGIN;
+import static io.netty.handler.codec.http.HttpHeaders.Names.CONNECTION;
+import static io.netty.handler.codec.http.HttpHeaders.Names.CONTENT_LENGTH;
+import static io.netty.handler.codec.http.HttpHeaders.Names.CONTENT_TYPE;
+import static io.netty.handler.codec.http.HttpHeaders.Names.LOCATION;
+import static io.netty.handler.codec.http.HttpHeaders.Values.CLOSE;
+import static io.netty.handler.codec.http.HttpMethod.GET;
+import static io.netty.handler.codec.http.HttpMethod.POST;
+import static io.netty.handler.codec.http.HttpMethod.PUT;
+import static io.netty.handler.codec.http.HttpResponseStatus.CONTINUE;
+import static io.netty.handler.codec.http.HttpResponseStatus.CREATED;
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
 import static org.apache.hadoop.hdfs.protocol.HdfsConstants.HDFS_URI_SCHEME;
 import static org.apache.hadoop.hdfs.security.token.delegation.DelegationTokenIdentifier.HDFS_DELEGATION_KIND;
 
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/FSImageHandler.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/FSImageHandler.java
index 9515a7e..eb93c87 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/FSImageHandler.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/FSImageHandler.java
@@ -22,21 +22,21 @@
 import java.util.List;
 import java.util.Map;
 
-import com.cloudera.io.netty.buffer.ByteBuf;
-import com.cloudera.io.netty.buffer.Unpooled;
-import com.cloudera.io.netty.channel.ChannelFutureListener;
-import com.cloudera.io.netty.channel.ChannelHandlerContext;
-import com.cloudera.io.netty.channel.SimpleChannelInboundHandler;
-import com.cloudera.io.netty.channel.group.ChannelGroup;
-import com.cloudera.io.netty.handler.codec.http.DefaultFullHttpResponse;
-import com.cloudera.io.netty.handler.codec.http.DefaultHttpResponse;
-import static com.cloudera.io.netty.handler.codec.http.HttpResponseStatus.*;
-
-import com.cloudera.io.netty.handler.codec.http.HttpMethod;
-import com.cloudera.io.netty.handler.codec.http.HttpRequest;
-import com.cloudera.io.netty.handler.codec.http.HttpResponseStatus;
-import static com.cloudera.io.netty.handler.codec.http.HttpVersion.*;
-import com.cloudera.io.netty.handler.codec.http.QueryStringDecoder;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.channel.group.ChannelGroup;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.DefaultHttpResponse;
+import static io.netty.handler.codec.http.HttpResponseStatus.*;
+
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import static io.netty.handler.codec.http.HttpVersion.*;
+import io.netty.handler.codec.http.QueryStringDecoder;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.apache.hadoop.hdfs.web.JsonUtil;
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/WebImageViewer.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/WebImageViewer.java
index 33a38cf..087972f 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/WebImageViewer.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/WebImageViewer.java
@@ -17,21 +17,21 @@
  */
 package org.apache.hadoop.hdfs.tools.offlineImageViewer;
 
-import com.cloudera.io.netty.bootstrap.ServerBootstrap;
-import com.cloudera.io.netty.channel.Channel;
-import com.cloudera.io.netty.channel.ChannelInitializer;
-import com.cloudera.io.netty.channel.ChannelPipeline;
-import com.cloudera.io.netty.channel.EventLoopGroup;
-import com.cloudera.io.netty.channel.group.ChannelGroup;
-import com.cloudera.io.netty.channel.group.DefaultChannelGroup;
-import com.cloudera.io.netty.channel.nio.NioEventLoopGroup;
-import com.cloudera.io.netty.channel.socket.SocketChannel;
-import com.cloudera.io.netty.channel.socket.nio.NioServerSocketChannel;
-import com.cloudera.io.netty.handler.codec.http.HttpRequestDecoder;
-import com.cloudera.io.netty.handler.codec.http.HttpResponseEncoder;
-import com.cloudera.io.netty.handler.codec.string.StringEncoder;
-import com.cloudera.io.netty.util.concurrent.GlobalEventExecutor;
 import com.google.common.annotations.VisibleForTesting;
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.group.ChannelGroup;
+import io.netty.channel.group.DefaultChannelGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.handler.codec.http.HttpRequestDecoder;
+import io.netty.handler.codec.http.HttpResponseEncoder;
+import io.netty.handler.codec.string.StringEncoder;
+import io.netty.util.concurrent.GlobalEventExecutor;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/TestParameterParser.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/TestParameterParser.java
index 79a9640..217d6b5 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/TestParameterParser.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/web/webhdfs/TestParameterParser.java
@@ -28,7 +28,7 @@
 import org.junit.Assert;
 import org.junit.Test;
 
-import com.cloudera.io.netty.handler.codec.http.QueryStringDecoder;
+import io.netty.handler.codec.http.QueryStringDecoder;
 
 import javax.servlet.ServletContext;
 
diff --git a/hadoop-mapreduce-project/hadoop-mapreduce-client/pom.xml b/hadoop-mapreduce-project/hadoop-mapreduce-client/pom.xml
index 7f96418..1f17c8f 100644
--- a/hadoop-mapreduce-project/hadoop-mapreduce-client/pom.xml
+++ b/hadoop-mapreduce-project/hadoop-mapreduce-client/pom.xml
@@ -130,17 +130,6 @@
       <scope>test</scope>
     </dependency>
     <dependency>
-      <groupId>org.apache.hadoop</groupId>
-      <artifactId>cloudera-netty-all</artifactId>
-      <exclusions>
-        <exclusion>
-          <groupId>io.netty</groupId>
-          <artifactId>netty-all</artifactId>
-        </exclusion>
-      </exclusions>
-      <scope>compile</scope>
-    </dependency>
-    <dependency>
       <groupId>io.netty</groupId>
       <artifactId>netty</artifactId>
     </dependency>
diff --git a/hadoop-mapreduce-project/pom.xml b/hadoop-mapreduce-project/pom.xml
index 1226f1b..ddcb511 100644
--- a/hadoop-mapreduce-project/pom.xml
+++ b/hadoop-mapreduce-project/pom.xml
@@ -64,10 +64,6 @@
           <artifactId>netty</artifactId>
         </exclusion>
         <exclusion>
-          <groupId>org.apache.hadoop</groupId>
-          <artifactId>cloudera-netty-all</artifactId>
-        </exclusion>
-        <exclusion>
           <groupId>org.apache.velocity</groupId>
           <artifactId>velocity</artifactId>
         </exclusion>
diff --git a/hadoop-project/pom.xml b/hadoop-project/pom.xml
index b3b6146..461c866 100644
--- a/hadoop-project/pom.xml
+++ b/hadoop-project/pom.xml
@@ -554,15 +554,15 @@
       </dependency>
 
       <dependency>
-        <groupId>org.apache.hadoop</groupId>
-        <artifactId>cloudera-netty-all</artifactId>
-        <version>${project.version}</version>
+        <groupId>io.netty</groupId>
+        <artifactId>netty</artifactId>
+        <version>3.6.2.Final</version>
       </dependency>
 
       <dependency>
         <groupId>io.netty</groupId>
-        <artifactId>netty</artifactId>
-        <version>3.6.2.Final</version>
+        <artifactId>netty-all</artifactId>
+        <version>4.0.23.Final</version>
       </dependency>
 
       <dependency>
-- 
1.7.9.5

